### PR TITLE
Rename Decision to WorkflowTask.

### PR DESCRIPTION
**Pull request by robholland**

### CODE REVIEW

File: lib/temporal.rb [hunk 1]
```diff
@@ -50,16 +50,16 @@ def signal_workflow(workflow, signal, workflow_id, run_id, input = nil)
       )
     end
 
-    def reset_workflow(namespace, workflow_id, run_id, decision_task_id: nil, reason: 'manual reset')
-      decision_task_id ||= get_last_completed_decision_task(namespace, workflow_id, run_id)
-      raise Error, 'Could not find a completed decision task event' unless decision_task_id
+    def reset_workflow(namespace, workflow_id, run_id, workflow_task_id: nil, reason: 'manual reset')
+      workflow_task_id ||= get_last_completed_workflow_task(namespace, workflow_id, run_id)
+      raise Error, 'Could not find a completed workflow_task task event' unless workflow_task_id
```
Comment (antstorm) [inline lib/temporal.rb:55]: nit: `workflow_task task`

Comment (robholland) [reply lib/temporal.rb]: Fixed.

File: lib/temporal/workflow/context.rb [hunk 2]
```diff
@@ -41,7 +41,7 @@ def execute_activity(activity_class, *input, **args)
 
         execution_options = ExecutionOptions.new(activity_class, options)
 
-        decision = Decision::ScheduleActivity.new(
+        workflow_task = WorkflowTask::ScheduleActivity.new(
```
Comment (antstorm) [inline lib/temporal/workflow/context.rb:44]: I think this is a wrong assumption — top level `Decision` is not the same as `DecisionTask`, so therefore it does not translate to `WorkflowTask`. This should be a `Command`, which is what a "decision" was used as a synonym of

File: lib/temporal/workflow/history/event.rb [hunk 3]
```diff
@@ -42,8 +42,8 @@ def initialize(raw_event)
         end
 
         # Returns the ID of the first event associated with the current event,
-        # referred to as a "decision" event. Not related to DecisionTask.
-        def decision_id
+        # referred to as a "workflow_task" event. Not related to WorkflowTaskTask.
```
Comment (antstorm) [inline lib/temporal/workflow/history/event.rb:45]: ^ This comment explain exactly what is wrong with the rename :)

`decision_id` should be `command_id`

**Review by antstorm — CHANGES_REQUESTED** [review]

**Comment by robholland** [comment]
Lets just scrap this, I'll file a fresh PR :)

**Pull request closed without merging**